docs/dev/morphology/morphology.rst

```diff
@@ -0,0 +1,131 @@
+.. _morphology_loading:
+
+Morphology loading in NEURON
+============================
+
+Traditionally reading morphologies with NEURON has been coupled to the ``Import3D GUI`` tool. 
+The ``Import3D GUI`` tool is a graphical tool that allows the user to load a morphology and then interactively modify it.
+:ref:`import3d_tutorial` provides a tutorial on how to use the ``Import3D GUI`` tool.
+
+The underlying implementation is in HOC and the GUI constructs can be disreagarded, since loading numerous morphologies for simulation does not make any use of them.
+
+
+Import3D_GUI Morphology Loading - Legacy
+----------------------------------------
+
+A HOC ``CellType`` needs to be instantiated and the Import3D stack will be used to load the morphology. 
+
+Please refer to `Cell.hoc from BlueBrain/neurodamus <https://github.com/BlueBrain/neurodamus/blob/8263f1d70995d5815c30691d2b14ef4a840ff356/core/hoc/Cell.hoc>`_ for an example of HOC ``CellType``.
+
+.. image:: Import3D-hoc.png
+
+The ``instantiate`` method of Import3D stack will create a list of HOC commands that will be executed to load the morphology.
+
+Given a simple single point soma simple Neurolucida morphology ::
+
+    ; This is the same morphology as simple.swc
+    ;
+    ;           (0, 5)
+    ; (-5, 5)----- ------ (6, 5)
+    ;             |
+    ;             |
+    ;             |
+    ;             | Type = 3
+    ;             |
+    ;             o origin
+    ;             |
+    ;             | Type = 2
+    ;             |
+    ;             |
+    ;(-5, -4)----- ------ (6, -4)
+    ;           (0, -4)
+
+
+    ("CellBody"
+    (Color Red)
+    (CellBody)
+    (0 0 0 2)
+    )
+
+    ((Dendrite)
+    (0 0 0 2)
+    (0 5 0 2)
+    (
+    (-5 5 0 3)
+    |
+    (6 5 0 3)
+    )
+    )
+
+
+    ((Axon)
+    (0 0 0 2)
+    (0 -4 0 2)
+    (
+    (6 -4 0 4)
+    |
+    (-5 -4 0 4)
+    )
+    )
+
+``instantiate`` will parse the file and create the following HOC commands
+that will be executed in the HOC interpreter, thus loading the morphology: ::
```

Comment:
If `instantiate` is called with the first argument being a Python object, no HOC commands are generated. Instead, it will invoke e.g. `neuron._create_sections_in_obj`, `neuron._connect_sections_in_obj`, `neuron._pt3dadd_in_obj` to load the cell into a Python object.

Reply:
Haven't really explored this Python side of the morphology loding, since we were exclusively relying on HOC on our side. Is it something mainstreamed?

Reply:
No idea how long instantiating inside HOC objects has been supported, but Python support was added ~7 years ago in 6a630ebf17797b283e54935e2dea9eba46820085, and this is how we've taught it ever since.

Reply:
I see. Well I guess this requires some consideration wrt the MorphIO integration as well.

For this PR, maybe you could add a small section and link (existing?) resources? Otherwise a follow-up PR shall be in order.

Reply:
So there's an example at: https://nrn.readthedocs.io/en/8.2.2/guide/bio_faq.html#how-do-i-work-with-neuron-morphologies

```python
from neuron import h
h.load_file("stdlib.hoc")
h.load_file("import3d.hoc")

class Pyramidal:
    def __init__(self):
        self.load_morphology()
        # do discretization, ion channels, etc
    def load_morphology(self):
        cell = h.Import3d_SWC_read()
        cell.input("c91662.swc")
        i3d = h.Import3d_GUI(cell, False)
        i3d.instantiate(self)

pyr = Pyramidal()
```

Reply:
Ok, I've linked that section